src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+                EnergyPlus::MixerComponent::GetZoneMixerIndex(state, zoneMixerName, zoneMixerIndex, zoneMixerErrFound, thisExhSys.Name);
+
+                if (!zoneMixerErrFound) {
+                    // With the correct MixerNum Initialize
+                    EnergyPlus::MixerComponent::InitAirMixer(state, zoneMixerIndex); // Initialize all Mixer related parameters
+
+                    // See if need to do the zone mixer's CheckEquipName() function
+                    ValidateComponent(state, "AirLoopHVAC:ZoneMixer", zoneMixerName, IsNotOK, "AirLoopHVAC:ExhaustSystem");
+                    if (IsNotOK) {
+                        ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                        ShowContinueError(state, "ZoneMixer Name =" + zoneMixerName + " mismatch or not found.");
+                        ErrorsFound = true;
+                    } else {
+                        // normal conditions
+                    }
+                } else {
+                    ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                    ShowContinueError(state, "ZoneMixer Name =" + zoneMixerName + "not found.");
+                    ErrorsFound = true;
+                }
+                thisExhSys.ZoneMixerName = zoneMixerName;
+                thisExhSys.ZoneMixerIndex = zoneMixerIndex;
+
+                std::string centralFanType = ip->getAlphaFieldValue(objectFields, objectSchemaProps, "fan_object_type");
+                int centralFanTypeNum = 0;
+                // 2022-01: Check fan types and gives warnings
+                if (UtilityRoutines::SameString(centralFanType, "Fan:SystemModel")) {
+                    centralFanTypeNum = DataHVACGlobals::FanType_SystemModelObject;
+                } else if (UtilityRoutines::SameString(centralFanType, "Fan:ComponentModel")) {
+                    centralFanTypeNum = DataHVACGlobals::FanType_ComponentModel;
+                } else {
+                    ShowSevereError(state, RoutineName + cCurrentModuleObject + "=" + thisExhSys.Name);
+                    ShowContinueError(state, "Fan Type =" + centralFanType + "not supported; ");
+                    ShowContinueError(state, "it needs to be either Fan:SystemModel or Fan:ComponentModel.");
+                    ErrorsFound = true;
+                }
+                thisExhSys.CentralFanTypeNum = centralFanTypeNum;
+
+                std::string centralFanName = ip->getAlphaFieldValue(objectFields, objectSchemaProps, "fan_name");
+                int availSchNum = 0;
+                int centralFanIndex = -1; // zero based
+                if (centralFanTypeNum == DataHVACGlobals::FanType_SystemModelObject) {
+                    // 2022-02-04: This type is zero indexed.
+                    // 2022-02-04: Need to process the System fan here first
+                    state.dataHVACFan->fanObjs.emplace_back(new HVACFan::FanSystem(state, centralFanName));
```

Comment:
Hmmmm. I have not seen us use arrays of object pointers before. We need to do more of this I think since it supports polymorphism. 👍

Reply:
👍 There seems to be very few example of using this way in the code. Thanks @mjwitte for digging out one example original for reference.